fe/fe-core/src/main/java/org/apache/doris/common/util/TimeUtils.java

```diff
@@ -305,4 +314,11 @@ public static LocalDateTime formatDateTimeAndFullZero(String datetime, DateTimeF
                 LocalTime.of(hour, minute, second, milliSecond * 1000000));
     }
 
+    private static String formatDateStr(String dateStr) {
+        String[] parts = dateStr.trim().split("[ :-]+");
+        return String.format("%s-%02d-%02d%s", parts[0], Integer.parseInt(parts[1]), Integer.parseInt(parts[2]),
+                parts.length > 3 ? String.format(" %02d:%02d:%02d", Integer.parseInt(parts[3]),
+                        Integer.parseInt(parts[4]), Integer.parseInt(parts[5])) : "");
+    }
+
```

Comment:
SimpleDateFormat will try to parse it into a valid datetime according to certain rules when the incoming string does not exactly match the specified date format. For example, there is only one month in "2013-2-01", and SimpleDateFormat will format the string into "2013-02-01" for parsing. But DateTimeFormatter has strict format validation. So a formatting logic for date strings is added here to complete the month and day in the date format.

Reply:
ok